### ISSUE TITLE
How to overwrite "dialog" with Sweet alert?

### ISSUE THREAD

**Issue by vpadhariya** [kartik-v/yii2-grid#757]
Hi Kartik,

how can I configure "SweetAlert" widget, I see you configure plugin to use "Confirm Dialog" whereas I am using this extension: https://github.com/yii2mod/yii2-sweet-alert and every where I am using sweet alert but I do not find a way to change the dialog to Sweet alert.

let me know, your help is much appreciated.

NOTE: this is for deleting confirmation alert.

**Comment by kartik-v** [comment]
You can cross check the docs... 

You can set `GridView::krajeeDialogSettings['useNative']` property to true to render native alerts or configure your own callback.

For ActionColumn you can entirely customize and create your own action buttons -  for the delete button the `data-confirm` property for the delete action button controls whether to show a dialog...

**Comment by vpadhariya** [comment]
Thank you Kartik.

I have completely overwrite "delete" button template and I am using "onclick" event, below is my code.

`[
	'class'          => 'yii\grid\ActionColumn',
	'headerOptions'  => ['class' => 'text-center'],
	'contentOptions' => ['class' => 'text-center'],
	'template'       => "{update} {delete}",
	'controller'     => '/VendorDashboard/event-item',
	'buttons'        => [
		'update'   => function($url, $model) {
			return Html::a('<span class="glyphicon glyphicon-pencil"></span>', ['/VendorDashboard/event-item/update', 'id' => $model->id, 'asDialog' => '1', 'edit_from' => 'event-view'], ['onclick'=>'return showModal("'. yii::t('app', 'Update Rental Item') . ' : ' . $model->vendorItem->name . '", $(this).attr("href"), 60, 850); return false;', 'title' => yii::t('app', 'Update Rental Item'), 'data-pjax' => '0', 'data-toggle' => 'tooltip']);
		},
		'delete' => function ($url, $model) {
			return Html::a('<span class="glyphicon glyphicon-trash"></span>', '#', [
				'title'      => Yii::t('yii', 'Delete'),
				'aria-label' => Yii::t('yii', 'Delete'),
				'onclick'    => "yii.confirm('" . Yii::t('app', 'Are you sure you want to delete this rental item?') . "', function(){
							$.ajax('$url', {type: 'POST'}).done(function(data) {
								$.pjax.reload('#items-in-event', {timeout : false}).done(function () {
									$.pjax.reload('#event-invoice-details', {timeout : false}).done(function () {
										$.pjax.reload('#main-alert-widget', {timeout : false});
									});
								});
							});
						});return false;",
			]);
		},
	]
]`

and below is my JS 
` yii.confirm = function (message, okCallback, cancelCallback) {
        swal({
            title: message,
            type: 'warning',
            showCancelButton: true,
            closeOnConfirm: true,
            allowOutsideClick: true
        }, okCallback);
};`

Using native bring back me to typical Browser base Confirm while I want to configure Sweet alert and I do not find any way how to configure them

**Comment by vpadhariya** [comment]
Not sure why code is not being formatted. but hope I am able to explain myself.

**Comment by vpadhariya** [comment]
OK got it thanks, I have to set like this.

`'krajeeDialogSettings' => ['overrideYiiConfirm' => false]` 

and my sweet alerts are working.
